PROTOCOL.md

```diff
@@ -1,183 +1,390 @@
 # Protocol
 
-### Requests sent by client
+## Data Types
+
+Our protocol is a JSON protocol. This means that we have the following types at our disposal: `string`, `number`, `object`, `array`, `boolean` and `null`.
+- In the following, the `String` type is the same as the JSON `string` type.
+- We also use an `Integer` type which represents a JSON `number` with the additional constraints that it needs to be a positive integer and that it must fit on a 32 bits representation.
```

Comment:
I would suggest to simplify the doc so that we directly write the JSON type name in the explainer tables, meaning:
- `string` instead of `String`
- `number` instead of `Integer` and `Long`
- etc.

Our protocol doesn't contain any fields where the `number`s would be non-integers, **that** could be said here in the only note we need:

> In the following, all `number` types are positive integers or zero.